main.tf

```
@@ -56,6 +56,7 @@ resource "azurerm_application_insights" "this" {
   location            = var.location
   application_type    = "web"
   workspace_id        = var.workspace_id
+  retention_in_days   = var.retention_in_days
```

We enforce **workspace-based** Application Insights in this module, which sends logs to Log Analytics. In this case, the retention must be set in the target Log Analytics workspace.

It might be that the App Insights component you're referring to has not yet migrated to workspace-spaced, which it should.

It would still be a good idea to explicitly set the value to `null` and add a comment about why 🙂 

```suggestion

  # Irrelevant when "workspace_id" is set.
  # Retention must be configured in Log Analytics workspace.
  retention_in_days = null
```

---

Would obviously require removing the `retention_in_days` variable as well.

---

This would also make this PR a refactor, since it's no longer adding a new feature, but still making a change to the code.

---

We have a workspace based App Insights, but will set it in Log Analytics if thats where it is 🙂